subsys/dfu/boot/mcuboot.c

```diff
@@ -361,40 +367,48 @@ static int boot_magic_code_check(const u32_t *magic)
 	return BOOT_MAGIC_UNSET;
 }
 
-static int boot_magic_state_read(u32_t bank_offs)
+static int boot_magic_state_read(u8_t bank_id)
 {
-	u32_t magic[4];
+	const struct flash_area *fa;
+	u32_t magic[(sizeof(u32_t) - 1 + BOOT_MAGIC_SZ) / sizeof(u32_t)];
```

Comment:
why? add comment in the code.

Reply:
BOOT_MAGIC_SZ is defined, I want it to be used instead of a fixed value 16

Reply:
I saw that.